### ISSUE TITLE
Use of uninitialized value $PRIO_PROCESS in concatenation (.) or string at /usr/bin/reniced line 395.

### ISSUE THREAD

**Issue by mmitch** [mmitch/reniced#3]
(detected here: https://github.com/mmitch/reniced/issues/2#issuecomment-840263333)

When BSD::Resource is installed, the following error occurs:

```
Use of uninitialized value $PRIO_PROCESS in concatenation (.) or string at /usr/bin/reniced line 395.
```

Looks like the `eval` line is wrong, as `$PRIO_PROCESS` gets evaluated in the string instead before evaluating the string:

```perl
sub get_prio_process()
# get the numerical value for PRIO_PROCESS
{
    # Check if BSD::Resource can be loaded
    eval { require BSD::Resource; };

    if (not $@) {
	# BSD::Resource which has the constant is available
	eval "$PRIO_PROCESS = BSD::Resource::PRIO_PROCESS;";
	debug 'PRIO_PROCESS set to %d via BSD::Resource', $PRIO_PROCESS;
    } else {
	# dirty fallback, works for my Linux 2.6.11 i686 GNU/Linux
	# see setpriority(2) and /usr/include/bits/resource.h
	$PRIO_PROCESS = 0;
	debug 'PRIO_PROCESS to %d via fallback', $PRIO_PROCESS;
    }
}
```

* Possible fix: use `'` instead of `"`.
* Big question: Is there no test to find this? I am pretty sure I ran the script with as well as witout `BSD::Resource` installed…

**Comment by mmitch** [comment]
@Priyanshu139 The problem is fixed in master - please download the newest version and test it (just overwrite the binary as before).

I'm still thinking about proper tests, but you can already have that fix :-)

**Comment by Priyanshu139** [comment]
@mmitch Thanks buddy :) it works like a charm now

**Comment by mmitch** [comment]
* commit 0706ed0 added proper tests (run once with `BSD::Resource` and once without)
* _reniced v1.23_ was released to roll out the fix from commit 153fc09
